src/app/parcels/ActionBar/StatusesModal.tsx

```diff
@@ -39,7 +39,7 @@ const StatusesBarModal: React.FC<StatusesBarModalProps> = (props) => {
             date
                 .set("year", newDate?.year() ?? date.year())
                 .set("month", newDate?.month() ?? date.month())
-                .set("day", newDate?.day() ?? date.day())
```

Comment:
.day sets the day of the week not the day of the month. So when you picked a date on the calendar, it would pick the wrong date in the same month & same day of the week.